src/main/java/seedu/address/logic/commands/RelateCommand.java

```diff
@@ -40,9 +40,13 @@ public RelateCommand(IdContainsDigitsPredicate predicate) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        // reset user view from any previous commands
+        model.clearFilter();
+
         // if ids are valid AND exists, model will display them, otherwise, it will be an empty list
         model.stackFilters(predicate);
```

Comment:
Can I suggest moving the
```
model.clearFilter();
model.stackFIlters(predicate);
```
to after line 65? I think it would be better as logically, we want the UI to only update after the completion of the action. Doing so also will prevent a `clearFilter()` -> `stackFilters(...)` -> `clearFilter()` happening when the ids are equal (although it is unnoticeable on the UI but it is happening).